### PR TITLE
Adds directory file streamer adapter

### PR DESCRIPTION
**Pull request by htollefsen**

### CODE REVIEW

**Comment by htollefsen** [comment]
@valisj Is this do-able in the current StreamAdapter?

File: src/Transfer/Commons/Stream/Adapter/DirectoryStreamAdapter.php [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    {
+        $this->path = $path;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function receive(Request $request)
+    {
+        $rii = new \RecursiveIteratorIterator(new \RecursiveDirectoryIterator($this->path));
```
Comment (valisj) [inline src/Transfer/Commons/Stream/Adapter/DirectoryStreamAdapter.php]: I think `path` should be included in the request. Like so:
``` php
$adapter->receive(Request::createWithHeaders(array('path' => '...')));

public function receive(Request $request) {
    $path = $request->getHeader('path');
}
```

File: src/Transfer/Commons/Stream/Adapter/DirectoryStreamAdapter.php [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+        $files = array();
+        foreach ($rii as $file) {
+            if ($file->isDir()) {
+                continue;
+            }
+            $stream = fopen($file->getPathname(), 'r');
+            $files[] = stream_get_contents($stream);
```
Comment (valisj) [inline src/Transfer/Commons/Stream/Adapter/DirectoryStreamAdapter.php]: A problem with this approach is that all of the files are loaded within the receive call, which might take some time depending on how many files there are. The receive method should instead return an iterator which will do the work of loading files when the `current()` function is called. See here: https://github.com/transfer-framework/transfer/blob/1.0/src/Transfer/Adapter/LocalDirectoryAdapter.php#L21

**Comment by htollefsen** [comment]
Closing, functionality already available.

**Pull request closed without merging**